simtools/data_model/metadata_collector.py

```diff
@@ -228,23 +248,46 @@ def _read_input_metadata_from_file(self):
 
         Raises
         ------
-        gen.InvalidConfigData
+        gen.InvalidConfigData, FileNotFoundError
             if metadata cannot be read from file.
 
         """
 
-        if self.args_dict is None or self.args_dict.get("input_meta", None) is None:
+        try:
+            metadata_file_name = (
+                self.args_dict.get("input_meta", None)
+                if metadata_file_name is None
+                else metadata_file_name
+            )
+        except TypeError:
+            pass
+
+        if metadata_file_name is None:
             self._logger.debug("No input metadata file defined.")
             return {}
 
-        try:
-            self._logger.debug(f"Reading meta data from {self.args_dict['input_meta']}")
-            _input_metadata = gen.collect_data_from_yaml_or_dict(
-                in_yaml=self.args_dict["input_meta"], in_dict=None
-            )
-        except (gen.InvalidConfigData, FileNotFoundError):
-            self._logger.error("Failed reading metadata from file.")
-            raise
+        # metadata from yml file
+        if Path(metadata_file_name).suffix == ".yml":
+            try:
+                self._logger.debug("Reading meta data from %s", metadata_file_name)
+                _input_metadata = gen.collect_data_from_yaml_or_dict(
+                    in_yaml=metadata_file_name, in_dict=None
+                )
+            except (gen.InvalidConfigData, FileNotFoundError):
+                self._logger.error("Failed reading metadata from %s", metadata_file_name)
+                raise
+        # metadata from table meta in ecsv file
+        elif Path(metadata_file_name).suffix == ".ecsv":
+            try:
+                _input_metadata = {observatory: Table.read(metadata_file_name).meta[observatory]}
+            except (FileNotFoundError, KeyError):
```

Comment:
In general my impression is that we are too acceptive with the errors without distinguishing between them. For example, FileNotFoundError and KeyError are pretty different errors, and it would help to deal with these cases separately, as we are not limited to one `except` statement.

Reply:
That is fine, the code does what it is supposed to, and I guess each case inside the if statements is already being tested, so all good.